Mathlib/Topology/Homeomorph.lean

```diff
@@ -701,6 +701,47 @@ def ulift.{u, v} {X : Type u} [TopologicalSpace X] : ULift.{v, u} X ≃ₜ X whe
   continuous_invFun := continuous_uLift_up
   toEquiv := Equiv.ulift
 
+/-- `Equiv.sumArrowEquivProdArrow` as a homeomorphism. The natural homeomorphism
+`(ι ⊕ ι' → X) ≃ₜ (ι → X) × (ι' → X)` -/
+@[simps!]
+def sumArrowHomeomorphProdArrow {ι ι' : Type*} : (ι ⊕ ι' → X) ≃ₜ (ι → X) × (ι' → X)  where
+  toEquiv := Equiv.sumArrowEquivProdArrow _ _ _
+  continuous_toFun := by
+    simp only [Equiv.sumArrowEquivProdArrow, Equiv.coe_fn_mk, continuous_prod_mk]
+    continuity
+  continuous_invFun := continuous_pi fun i ↦ match i with
+    | .inl i => by apply (continuous_apply _).comp' continuous_fst
+    | .inr i => by apply (continuous_apply _).comp' continuous_snd
+
+private theorem _root_.Fin.appendEquiv_eq_Homeomorph (m n : ℕ) : Fin.appendEquiv m n =
```

Comment:
I feel like this could be nicer using `isHomeomorph_iff_exists_homeomorph` but that is further down the file. Maybe there could be a lemma like this for constructing `Homeomorph`s from `Equiv`s?

Reply:
I tend to agree, but if you want to try that out, you should do so in another PR (probably after this one since this one is almost ready)